src/app/clusters/operational-credentials/operational-credentials.cpp

```diff
@@ -33,33 +36,193 @@
+    //  -1: Read/Write the whole list content, including the number of elements in the list
+    //   0: Read/Write the number of elements in the list, represented as a uint16_t
+    //   n: Read/Write the nth element of the list
+    //
+    // Since the first 2 bytes of the attribute are used to store the number of elements, elements indexing starts
+    // at 1. In order to hide this to the rest of the code of this file, the element index is incremented by 1 here.
+    // This also allows calling writeAttribute() with no index arg to mean "write the length".
+    
+    return emAfReadOrWriteAttribute(&record, NULL, buffer, 0, true, index + 1);
+}
+
+EmberAfStatus writeFabric(FabricId fabricId, NodeId nodeId, uint16_t vendorId, int32_t index)
+{
+    EmberAfStatus status    = EMBER_ZCL_STATUS_SUCCESS;
+    AttributeId attributeId = ZCL_FABRICS_ATTRIBUTE_ID;
+
+    EmberAfFabricDescriptor fabricDescriptor;
+    fabricDescriptor.FabricId = fabricId;
+    fabricDescriptor.NodeId   = nodeId;
+    fabricDescriptor.VendorId = vendorId;
+
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Writing admin into attribute store at index %d: fabricId %" PRIX64 
+                            ", nodeId %" PRIX64 " vendorId %" PRIX16, index, fabricId, nodeId, vendorId);
+    status =  writeFabricAttribute(0, attributeId, (uint8_t *) &fabricDescriptor, index);
+    return status;
+}
+
+CHIP_ERROR writeAdminsIntoFabricsListAttribute(void)
+{
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Call to writeAdminsIntoFabricsListAttribute");
+
+    // Loop through admins
+    auto pairing = GetGlobalAdminPairingTable().cbegin();
+    int32_t fabricIndex = 0;
+    while (pairing != GetGlobalAdminPairingTable().cend())
+    { 
+        NodeId nodeId = pairing->GetNodeId();
+        uint64_t fabricId = pairing->GetFabricId();
+        uint16_t vendorId = pairing->GetVendorId();
+
+        // Skip over uninitialized admins
+        if (nodeId != kUndefinedNodeId && fabricId != kUndefinedFabricId && vendorId != kUndefinedVendorId)
+        {
+            if (writeFabric(fabricId, nodeId, vendorId, fabricIndex) != EMBER_ZCL_STATUS_SUCCESS)
+            {
+                emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Failed to write admin with fabricId %" PRIX64 " in fabrics list", fabricId);
+            }
+            fabricIndex ++;
+        }
+        pairing++;
+    }
+
+    // Store the count of fabrics we just stored
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Stored %" PRIX32 " admins in fabrics list attribute.", fabricIndex);
+    if (writeFabricAttribute(0, ZCL_FABRICS_ATTRIBUTE_ID, (uint8_t *) &fabricIndex) != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Failed to write admin count %" PRIX32 " in fabrics list", fabricIndex);
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+AdminPairingInfo * retrieveCurrentAdmin()
+{
+    uint64_t fabricId = emberAfCurrentCommand()->source;
```

Comment:
This doesn't look right.  The source of a command is the peer node id.  The fabric is not stored in a command at all right now, afaict.

That fact that all these things are just ints, so we can assign FabricId to a NodeId or vice versa, is deeply unfortunate.

Reply:
This works, this command retrieves the commissioner node id which I am temporarily using to identify my current admin. In the future I'd want to use the fabric index.

Reply:
This is making a lot of assumptions (e.g. that the command source is the same as the commissioner, etc).  This needs some very clear comments explaining the setup, probably along with a pointer to the issue that's filed to make this do something different.

Reply:
added documentation and TODOs.